crates/sol-types/src/types/struct.rs

```diff
@@ -63,9 +63,29 @@ pub trait SolStruct: 'static {
         self.tokenize().total_words() * Word::len_bytes()
     }
 
+    /// Returns component EIP-712 types. These types are used to construct
+    /// the `encodeType` string. These are the types of the struct's fields,
+    /// and should not include the root type.
+    fn eip712_components() -> Vec<Cow<'static, str>>;
```

Comment:
should this even be exposed? are there any other use cases

Reply:
unless we go through some extra trait wrangling, we can't both
- use it in the default impl of `eip712_encode_type`
- not expose it

Is there a specific setup you'd prefer?